### ISSUE TITLE
inaccurate zooming of component

### ISSUE THREAD

**Issue by rohittcodes** [seveibar/use-mouse-matrix-transform#2]
When you try to zoom in on a particular mouse location, it doesn't zoom into the correct location. Instead, it moves around and causes the mouse scroll to work inappropriately.

ref: https://github.com/tscircuit/snippets/issues/175

**Comment by rohittcodes** [comment]
Hey @seveibar, I've worked around the issue and have tried to reproduce the same. Here's the workaround

https://github.com/user-attachments/assets/e99f537a-f085-443a-beac-de72f66796df

But, there is an issue with stylings, it seems to be when the top and left values are increased (i.e., offset), causing the zoom to behave incorrectly. This could be happening because the inner div is positioned absolutely within a parent container that has additional offsets (like marginTop or marginLeft). I've tried implementing the same to issue [here](https://github.com/tscircuit/snippets/issues/175), by creating a temporary file and it's working well.

is it good to raise a pr for now?
